test/tests/functional/pbs_qrun.py

```diff
@@ -128,14 +129,13 @@ def test_qrun_hangs(self):
             except PbsRunError as e:
                 self.logger.info("Runjob throws error: " + e.msg[0])
         else:
-            out = self.scheduler.log_match(
-                "Starting Scheduling Cycle",
-                allmatch=True, interval=5, starttime=now, max_attempts=10)
-            if len(out) < 2:
+            try:
+                self.scheduler.log_match("Starting Scheduling Cycle",
+                                         interval=5, starttime=now,
+                                         max_attempts=10)
+                self.logger.info("No hangs. Parent process exit")
+            except PtlLogMatchError:
                 os.kill(pid, signal.SIGKILL)
                 os.waitpid(pid, 0)
                 self.logger.info("Runjob hung. Child process exit.")
-                self.assertFalse(True,
-                                 "Qrun didn't start another sched cycle")
-            else:
-                self.logger.info("No hangs. Parent process exit")
+                self.fail("Qrun didn't start another sched cycle")
```

Comment:
you are failing the test if log_match() is successful? This line should be part of the except block right?

Reply:
d'oh! you removed the else block completely, my bad